frontend/app/services/subscription-service.server.ts

```diff
@@ -102,5 +102,42 @@ function createSubscriptionService() {
     }
   }
 
-  return { getSubscription, updateSubscription };
+  async function validateConfirmationCode(userEmail: string, enteredConfirmationCode: string, sin: string) {
+    const auditService = getAuditService();
+    const instrumentationService = getInstrumentationService();
+
+    auditService.audit('alert-subscription.validate', { sin });
+
+    const dataToPass = {
+      email: userEmail,
+      confirmationCode: enteredConfirmationCode,
+    };
+    // TODO: "IT-Security won't like SIN being passed as identifier"
+    // TODO: add CDCP_API_BASE_URI
+    const url = new URL(`https://api.example.com/v1/users/codes/verify`);
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(dataToPass),
+    });
+
+    instrumentationService.countHttpStatus('http.client.cdcp-api.alert-confirm.posts', response.status);
```

Comment:
Convention in the application is to use the API endpoint as the name:
```suggestion
    instrumentationService.countHttpStatus('http.client.cdcp-api.codes.verify.posts', response.status);
```

Reply:
changed